### ISSUE TITLE
Send Keys is sending keys out side the container, But working in interactive mode via VNC

### ISSUE THREAD

**Issue by piyugit** [zalando/zalenium#957]
## 🐛 Bug Report

Send Keys is sending keys out side the container when not using intractive mode via VNC, But working in interactive mode via VNC it is entering path of file as well enter key to browse data.

## To Reproduce

- I have a web application where
- I am uploading any file by entering path of file ( already mounted folder/file )
- When I am entering path of mounted folder through 
  SendKeys.SendWait(filePath); 
 or sending enter through
  SendKeys.SendWait(@"{Enter}"); 

- It is entering path outside the containers in left bottom area.
![image](https://user-images.githubusercontent.com/31278532/56975832-3da17e80-6b8f-11e9-8353-c0eb199911e8.png)

/home/asaxena/deploymentdir is my loaclal directory and it is mounted on tmp/node/home/seluser/deploymentdir

I have setup zalenium on linux machine with following command :
docker run --rm -ti --name zalenium -p 4444:4444 -v /var/run/docker.sock:/var/run/docker.sock -v /temp/videos:/home/seluser/videos -v /home/asaxena/deploymentdir:/tmp/node/home/seluser/deploymentdir --privileged dosel/zalenium start --desiredContainers 6 --maxDockerSeleniumContainers 6 --maxTestSessions 15 --timeZone "Asia/Kolkata"

## Expected behavior
Enter keys and Send keys should work with in container when not interacting through VNC.

## Test script reproducing this issue (when applicable)

 public void UploadPdfTemplate(string filePath)
        {
            _parallelConfig.Driver.WaitForAction();
            SendKeys.SendWait(filePath);
            SendKeys.SendWait(@"{Enter}");
            _parallelConfig.Driver.WaitForAction(1);
        }
my path is : 10.104.5.160/PDF Templates/templateName

## Environment

OS:  Windows 10 and zalenium installed on Ubuntu 16.5
Zalenium Image Version(s): latest
Docker version: latest
Docker-Compose version (if applicable):
Exact Docker command to start the containers (if using docker-compose, provide 
the docker-compose file as well):
docker run --rm -ti --name zalenium -p 4444:4444 -v /var/run/docker.sock:/var/run/docker.sock -v /temp/videos:/home/seluser/videos -v /home/asaxena/deploymentdir:/tmp/node/home/seluser/deploymentdir --privileged dosel/zalenium start --desiredContainers 6 --maxDockerSeleniumContainers 6 --maxTestSessions 15 --timeZone "Asia/Kolkata"

**Comment by piyugit** [comment]
@diemol  please look in to this.

**Comment by diemol** [comment]
@piyugit how can I reproduce it with just this piece of code?
```
public void UploadPdfTemplate(string filePath)
{
_parallelConfig.Driver.WaitForAction();
SendKeys.SendWait(filePath);
SendKeys.SendWait(@"{Enter}");
_parallelConfig.Driver.WaitForAction(1);
}
my path is : 10.104.5.160/PDF Templates/templateName
```

Can you please provide a full functioning test? Also with an html page or public website.

**Comment by piyugit** [comment]
Hi @diemol  ,

this is not specific to any piece of code.
I am giving u a url of a public demo site :- https://encodable.com/uploaddemo/

- when you click on Chose File Button now the method starts:-

public void UploadPdfTemplate(string filePath)
{
_parallelConfig.Driver.WaitForAction();
SendKeys.SendWait(filePath);
SendKeys.SendWait(@"{Enter}");
_parallelConfig.Driver.WaitForAction(1);
}

- you can pass any path like your download folder where upload any file.

- the issue is with open popup where we donnt have any id of file path and open button.
we just use SendKeys to enter path and Enter to open.

- can you enter path and click on open/browse button by any mean that can work in any language like C#/java with salenium ( without using interacting through VNC because in VNC mode this code is working)

- the main issue is for single instance/container it will enter keys vale but when you have many containers it get confused and try to enter keys out side the containers.

**Comment by piyugit** [comment]
@diemol @elgalu  any thing more you are looking for ?

**Comment by diemol** [comment]
We need a complete piece of code to reproduce the issue, isolated from any frameworks you might be using, written instructions are not really helpful. The easiest way would be to provide a GitHub repo link with instructions on how to run you example that reproduces the issue.

**Comment by piyugit** [comment]
@diemol Due to security reasons I cant share original repo , but I have created test project on my public git for your reference url is : https://github.com/piyugit/ZeleniumTestRepo.git
this is a .net project.
please replce wd/hub url as per your url.
run this test on container it will fail but on Intractive mode it will pass.
let me know if more info required.

**Comment by piyugit** [comment]
@diemol any chance you have gone through the repo?

**Comment by diemol** [comment]
I have OSX and I need to install VIsual Studio to check that, I'll take some time.

**Comment by diemol** [comment]
Are you trying to do a `sendKeys` to this window?

![image](https://user-images.githubusercontent.com/5992658/57120622-f07e0200-6d72-11e9-8d18-fa85d7a5a79b.png)

If so, that is not possible, that is not part of the website, it is an operating system window and Selenium cannot interact with them.

**Comment by piyugit** [comment]
@diemol Thanks for your reply  : Yes I am trying to send keys on this windows, if i am sending keys in interactive mode through VNC on zalenium-docker  container i am able to send it. but once i start execution in parallel it is trying to send keys but out side the container.
Need not to install Visual studio u just use the same Url and any frame work or IDe.

**Comment by piyugit** [comment]
@diemol you can create or add this piece of code to upload/browse any file in any language or IDE like Java selenium , My aim is to upload file on given url by selecting file folder mounted on zalenium /docker machine , can you please try this and let me know if we can do file upload/Browse in parallel.

**Comment by micheletest** [comment]
@piyugit as @diemol mentions you cannot send keys to an operating system window with selenium. The reason it works in interactive mode is because you are taking control and navigating to the operating system window. This isn't a selenium or zalenium issue, it's just not possible to work in a non-browser environment. 

Selenium does allow you to send keys to a browser input field for a file upload. To fix your script, instead of clicking to open the file dialog, you can SendKeys to `uploadFile` the `filePath`. There are a lot of references and examples for selenium file upload with send keys if you search for that if you need more information.

**Comment by diemol** [comment]
Exactly, thank you @micheletest for writing the reasoning of it.
